### PR TITLE
Board valid placement

### PR DESCRIPTION
**Pull request by NickEdwin**
This should be the last of testing for valid placements!

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -24,4 +24,52 @@ def initialize(size = 16)
...
+  def column_not_diagonal?(coordinates)
+    column = []
+    coordinates.each_with_index do |coordinate, index|
+      next if coordinates[index + 1] == nil
+      column << (coordinate[1] == coordinates[index + 1][1])
+    end
+    column.all?(true)
+  end
+
+  def valid_placement?(ship, coordinates)
```
Comment (EHale64) [inline lib/board.rb:64]: Strong potential for refactor using unless logic
return false unless ship.length == coordinates.length
return false unless numbers_consecutive?(coordinates) || letters_consecutive?(coordinates)
return false unless row_not_diagonal?(coordinates) || column_not_diagonal?(coordinates)
true

Comment (NickEdwin) [reply lib/board.rb:64]: Makes sense, its shorter and easier on the eyes

File: lib/board.rb [hunk 2]
```diff
@@ -24,4 +24,52 @@ def initialize(size = 16)
   def valid_coordinate?(coordinate)
      @cells.has_key?(coordinate)
   end
+
+  def numbers_consecutive?(coordinates)
+    range = 1..4
+    numbers = coordinates.map do |coordinate|
```
Comment (EHale64) [inline lib/board.rb:30]: Potential to make our consecutive? methods into one line syntax.

File: test/board_test.rb [hunk 3]
```diff
@@ -27,4 +29,51 @@ def test_it_can_validate_coordinates
...
+    refute @board.letters_consecutive?(["A1", "B1", "D1"])
+    assert @board.letters_consecutive?(["A1", "B1", "C1"])
+    refute @board.letters_consecutive?(["B1", "A1"])
+    assert @board.letters_consecutive?(["C1", "D1"])
+  end
+
+  def test_it_finds_consecutive_coordinates
+    refute @board.valid_placement?(@cruiser, ["A1", "A2", "A4"])
+    assert @board.valid_placement?(@cruiser, ["A2", "A3", "A4"])
+    refute @board.valid_placement?(@submarine, ["B1", "B3"])
```
Comment (NickEdwin) [inline test/board_test.rb:58]: Ask about commenting out past tests that work to shorten length of code and potentially increase run speed.

**Pull request merged**